File: model/connection.py
import mysql.connector
from mysql.connector import Error


class Conection:
    def __init__(self):
        try:
            self.conn = mysql.connector.connect(
                host = "localhost",
                user = "root",
                password = "",
                database = "ventas_2025"
            )
            if not self.conn.is_connected():
                raise Error("No se pudo establecer la conexión.")
            self.cursor = self.conn.cursor()
            self.commit = self.conn.commit()
        except Error as e:
            print(f"Error al conectar con la base de datos: {e}")
            self.conn = None
            self.cursor = None

File: model/Connection.php
<?php
class Connection {
    protected $conn;
    protected $cursor;

    public function __construct() {
        try {
            $host = "localhost";
            $dbname = "ventas_2025";
            $username = "root";
            $password = "";

            $this->conn = new PDO("mysql:host=$host;dbname=$dbname", $username, $password);
            $this->conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
            $this->cursor = $this->conn;
        } catch(PDOException $e) {
            die("Connection failed: " . $e->getMessage());
        }
    }

    public function __destruct() {
        $this->conn = null;
        $this->cursor = null;
    }
}
?>